cpl/module_cplscalars.F90

```diff
@@ -0,0 +1,204 @@
+!> @file
+!> @brief Manage cpl_scalars
+!> @author mvertens@ucar.edu
+!> @author modified for FV3atm by Denise.Worthen@noaa.gov @date 03-03-2024
+
+!> Manage scalars in import and export states. Called by realizeConnectedCplFields
+!> to set the required scalar data into a state. The scalar_value will be set into
+!> a field with name flds_scalar_name. The scalar_id identifies which dimension in
+!> the scalar field is given by the scalar_value. The number of scalars is used to
+!> ensure that the scalar_id is within the bounds of the scalar field
+
+module module_cplscalars
+
+  use NUOPC
+  use ESMF, only : ESMF_Field, ESMF_Distgrid, ESMF_Grid, ESMF_State
+  use ESMF, only : ESMF_VM, ESMF_VMGetCurrent, ESMF_VMGet, ESMF_VMBroadCast
+  use ESMF, only : ESMF_LogFoundError, ESMF_LOGERR_PASSTHRU, ESMF_FAILURE, ESMF_SUCCESS
+  use ESMF, only : ESMF_LOGMSG_INFO, ESMF_LOGWRITE, ESMF_TYPEKIND_R8, ESMF_KIND_R8
+  use ESMF, only : ESMF_GridCreate, ESMF_FieldCreate, ESMF_StateGet, ESMF_DistGridCreate
+  use ESMF, only : ESMF_FieldGet
```

Comment:
In fv3atm we do not require that each symbol is explicitly specified using 'only' attribute for third-party libraries like ESMF. But I'm okay with it, if you prefer this style.